presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/client/AwsRequestSigner.java

```diff
@@ -84,32 +86,55 @@ public void process(final HttpRequest request, final HttpContext context)
             }
         }
 
-        DefaultRequest<?> awsRequest = new DefaultRequest<>(SERVICE_NAME);
-
         HttpHost host = (HttpHost) context.getAttribute(HTTP_TARGET_HOST);
         if (host != null) {
-            awsRequest.setEndpoint(URI.create(host.toURI()));
-        }
-        awsRequest.setHttpMethod(HttpMethodName.fromValue(method));
-        awsRequest.setResourcePath(uri.getRawPath());
-        awsRequest.setContent(content);
-        awsRequest.setParameters(parameters);
-        awsRequest.setHeaders(headers);
-
-        signer.sign(awsRequest, credentialsProvider.getCredentials());
-
-        Header[] newHeaders = awsRequest.getHeaders().entrySet().stream()
-                .map(entry -> new BasicHeader(entry.getKey(), entry.getValue()))
-                .toArray(Header[]::new);
-
-        request.setHeaders(newHeaders);
-
-        InputStream newContent = awsRequest.getContent();
-        checkState(newContent == null || request instanceof HttpEntityEnclosingRequest);
-        if (newContent != null) {
-            BasicHttpEntity entity = new BasicHttpEntity();
-            entity.setContent(newContent);
-            ((HttpEntityEnclosingRequest) request).setEntity(entity);
+            URI endpoint = URI.create(host.toURI());
+            SdkHttpFullRequest.Builder requestBuilder = SdkHttpFullRequest.builder()
+                    .method(SdkHttpMethod.fromValue(method))
+                    .uri(endpoint.resolve(uri.getRawPath()))
+                    .headers(headers.entrySet().stream()
```

Comment:
Rather than transforming the headers map again here, I would suggest creating it with the required type from the start, as it's created within this method itself.